skimage/morphology/_misc_cy.pyx

```diff
@@ -0,0 +1,122 @@
+#cython: cdivision=True
+#cython: boundscheck=False
+#cython: nonecheck=False
+#cython: wraparound=False
+
+"""Cython code used in `remove_near_objects` function."""
+
+cimport numpy as cnp
+
+from .._shared.fused_numerics cimport np_anyint
+
+
+def _remove_near_objects(
+    np_anyint[::1] out not None,
+    Py_ssize_t[::1] border_indices not None,
+    Py_ssize_t[::1] inner_indices not None,
+    kdtree,
+    cnp.float64_t p_norm,
+    cnp.float64_t min_distance,
+    tuple shape,
+):
+    """Remove objects until a minimal distance is ensured.
+
+    Iterates over all objects (pixels that aren't zero) inside an image and
+    removes "nearby" objects until all remaining ones are spaced more than a
+    given minimal distance from each other.
+
+    Parameters
+    ----------
+    out :
+        An array with labels for each object in `image` matching it in shape.
+    border_indices, inner_indices :
+        Indices into `out` for the border of objects (`border_indices`) and
+        the inner part of objects (`inner_indices`). `border_indices`
+        determines the iteration order; objects that are indexed first are
+        preserved. Indices must be sorted such, that indices pointing to the
+        same object are next to each other.
+    kdtree : scipy.spatial.cKDTree
```

Comment:
Ah, realized that this couldn't possibly be a public function, so take the above with a grain of salt.